### ISSUE TITLE
WCAG: Heading levels do not reflect content hierarchy

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#351]
### Summary

**Impact**: medium 

Global template issue – Heading hierarchy is not able to be controlled by page/site builder (pre-determined heading markup breaks heading hierarchy) 

Via Vision Australia assessment: August 2024

Note: DTA have a 90 day remediation period to address the identified issues within the audit (issue 11/35)

### Steps to reproduce

[digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/) (https://www.digital.gov.au) 

Throughout the site, headings have been marked up with levels that do not reflect the content hierarchy. On the homepage for example, The text "Featured Initiatives" has been marked up as a level 3 heading, making it a sub heading of the level 2 "Enabling digital government" heading, however, it should be a heading of the same level.  

The text "AI in government policy" is structurally a sub-heading of the preceding level 3 heading "Featured initiatives", however it has been marked up as a level 2 heading.

### Code used 

<h1 class="ct-heading ct-theme-dark ct-banner__title"> 

digital.gov.au beta 

</h1> 

…  

<h2> 

Enabling digital government 

</h2> 

…  

<h4 class="ct-heading ct-theme-light ct-promo-card__title"> 

Digital Experience Policy 

</h4> 

…  

<h4 class="ct-heading ct-theme-light ct-promo-card__title"> 

Digital architecture 

</h4> 

…  

<h3 class="ct-heading ct-theme-light ct-slider__title"> 

Featured initiatives 

</h3> 

…  

<h2 class="ct-heading ct-theme-light ct-slide__title"> 

AI in government policy 

</h2> 

### Why this matters 

Headings are a primary navigational tool for screen reader users and provide important information about content structure and architecture. Screen reader users can also use shortcuts to skip between headings on a screen. 

When headings do not have an appropriate hierarchical relationship with other headings on the page, they do not accurately represent the content's structure to users. This can make content difficult to understand and navigate.  

### Recommendation 

Mark up headings with a level that is appropriate to the page architecture. This should offer the page/site builder the option to choose heading levels on each component so they can build the page/site that reflects WCAG best practice for heading hierarchy.

In this instance, the promo card headings (<h4> elements) should be made <h3> elements so that heading levels are not skipped. The "Featured initiatives" heading should then be made a <h2> element. The "AI in government policy" heading should be updated to a <h3> element. 

Code example 

<h1 class="ct-heading ct-theme-dark ct-banner__title"> 

digital.gov.au beta 

</h1> 

…  

<h2> 

Enabling digital government 

</h2> 

…  

<h3 class="ct-heading ct-theme-light ct-promo-card__title"> 

Digital Experience Policy 

</h3> 

…  

<h3 class="ct-heading ct-theme-light ct-promo-card__title"> 

Digital architecture 

</h3> 

…  

<h2 class="ct-heading ct-theme-light ct-slider__title"> 

Featured initiatives 

</h2> 

…  

<h3 class="ct-heading ct-theme-light ct-slide__title"> 

AI in government policy 

</h3>

**Comment by fionamorrison23** [comment]
@CJE001  this is definitely an issue, and has been reported by another user (once I find that ticket I may close one and combine the info). 
This will require some rework on designs, and will take a fair bit of effort to get it right. There is also a question around what to do when we don't know how a user is going to use a particular component.

**Comment by AlexSkrypnyk** [comment]
Can this please be prioritised. Every site reports this issue.

The underlying issue is that the heading of the card (each card type) was implemented using a `heading` component at a specific `level` to replicate the design which used `Heading 4` font size.

The solution here is to decouple card's heading from the content headings completely as these are absolutely unrelated elements: the cards titles may look like heading 4, but they are not structurally related to content headings.

This change needs to take place in Figma and in UIKit.

The implementation in UIKit should use a `<div>` or other non-`<h*>` element and the styles should be inbuilt into the card itself not relying on the `heading` component.

**Comment by richardgaunt** [comment]
This has been addressed with the card updates completed in #695

**Comment by CJE001** [comment]
This is a global template issue (not just related to Promo Cards). Throughout the site, headings have been marked up with levels that do not reflect the content hierarchy, site builders require the ability to set/choose heading levels for each component (currently they're hard coded without ability to change). Recommend this issue be reopened.
